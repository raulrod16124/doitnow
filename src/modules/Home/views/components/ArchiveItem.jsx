import React, { useRef } from "react";
import { useDispatch } from "react-redux";

import { OpenConfirmationPropmt } from "../../../global/ConfirmationPropmt/state/actions";
import { DeleteTask, UpdateTask } from "../../../Home/state/actions";
import { Tag } from "../../../Home/views/components/Tag";

export const ArchiveItem = ({ item }) => {
  const dispatch = useDispatch();

  const menuResponsiveRef = useRef();

  const handleUnarchiveCompletedTask = (item) => {
    item.status = "done";
    dispatch(
      OpenConfirmationPropmt({
        message: `Do you want to unarchive the task ${item.title} ?`,
        acceptButton: "Unarchive",
        handleAccept: () => UpdateTask(item.id, item),
      })
    );
  };

  const handleDeleteItemById = (item) => {
    dispatch(
      OpenConfirmationPropmt({
        message: `Do you want to delete the task ${item.title} ?`,
        acceptButton: "Delete",
        handleAccept: () => DeleteTask(item.id),
      })
    );
  };

  const handleToggleMenuResponsive = (param) => {
    if (param === "open") {
      menuResponsiveRef.current.style.display = "flex";
      setTimeout(() => {
        menuResponsiveRef.current.style.height = "auto";
        menuResponsiveRef.current.style.width = "60%";
      }, 100);
    } else {
      menuResponsiveRef.current.style.display = "none";
      setTimeout(() => {
        menuResponsiveRef.current.style.height = "0%";
        menuResponsiveRef.current.style.width = "0%";
      }, 200);
    }
  };

  return (
    <li
      className="item archive-item"
      onClick={(e) => {
        if (
          e.target.className !== "item-menu-responsive" &&
          e.target.className !== "options" &&
          e.target.className !== "option" &&
          e.target.className !== "fas fa-ellipsis-v icon-item-settings"
        ) {
          handleToggleMenuResponsive();
        }
      }}
    >
      <h3 className="item-title">
        {item.title} <span className="item-date">{item.date}</span>
        <i
          class="fas fa-ellipsis-v icon-item-settings"
          onClick={() => handleToggleMenuResponsive("open")}
        ></i>
      </h3>
      <p className="item-description">{item.description}</p>
      <div className="content-tags">
        {item.tags &&
          item.tags.length > 0 &&
          item.tags.map((tag) => {
            return <Tag key={tag.tag} tag={tag} />;
          })}
      </div>
      <div className="item-footer">
        <p className="item-level">{item.level}</p>
        <div className="content-icons" ref={menuResponsiveRef}>
          <i
            className="fas fa-archive icon"
            onClick={() => handleUnarchiveCompletedTask(item)}
          ></i>
          <i
            className="fas fa-trash icon"
            onClick={() => handleDeleteItemById(item)}
          ></i>
        </div>
        <span className="item-date-responsive">{item.date}</span>
      </div>
      <div className="item-menu-responsive" ref={menuResponsiveRef}>
        <ul className="options">
          {item.status === "archive" && (
            <p
              className="option"
              onClick={() => {
                handleUnarchiveCompletedTask(item);
                handleToggleMenuResponsive();
              }}
            >
              Unarchive
            </p>
          )}
          <p
            className="option"
            onClick={() => {
              handleDeleteItemById(item);
              handleToggleMenuResponsive();
            }}
          >
            Delete
          </p>
        </ul>
      </div>
    </li>
  );
};
