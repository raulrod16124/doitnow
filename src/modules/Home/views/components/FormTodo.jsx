import React, { useEffect, useRef, useState } from "react";
import Calendar from "react-calendar";
import { CirclePicker } from "react-color";
import { useDispatch } from "react-redux";

import { Button } from "../../../../stories/Button";
import { Message } from "../../../global/Message";
import { CreateTask, UpdateTask } from "../../state/actions";
import { Tag } from "./Tag";

export const FormTodo = ({
  handleGetVisibilityFormState,
  itemToEdit = undefined,
}) => {
  const initialTodoState = {
    title: "",
    level: "easy",
    status: "todo",
    description: "",
    tags: [],
    date: new Date().toLocaleDateString(),
  };

  const dispatch = useDispatch();

  // Todo state
  const [todo, setTodo] = useState(initialTodoState);

  const tagRef = useRef();

  const [showColors, setshowColors] = useState({
    visibility: false,
    color: "#63CFE0",
  });

  useEffect(() => {
    setTodo(initialTodoState);
    if (itemToEdit !== undefined) {
      setTodo(itemToEdit);
    } else {
      setTodo(initialTodoState);
    }
  }, []);

  // Error State
  const [error, setError] = useState({
    visible: false,
    title: "Title is required",
  });

  // References
  const [Timestamp, setTimestamp] = useState(new Date());
  const [
    timestampSelectorVisibility,
    setTimestampSelectorVisibility,
  ] = useState(false);

  const handleGetStartDate = (e) => {
    setTimestampSelectorVisibility(false);
    // console.log(e.toLocaleDateString());
    setTimestamp(e);
    setTodo({ ...todo, date: e.toLocaleDateString() });
  };

  const handleSaveTodo = (e) => {
    e.preventDefault();
    if (todo.title !== "") {
      const userOwner = JSON.parse(localStorage.getItem("user"));
      const newTask = {
        title: todo.title,
        level: todo.level,
        status: todo.status,
        description: todo.description,
        date: todo.date,
        tags: todo.tags,
        owner: userOwner.email,
      };
      setTimestamp(new Date());
      if (itemToEdit !== undefined) {
        dispatch(UpdateTask(todo.id, newTask));
      } else {
        dispatch(CreateTask(newTask));
      }
      setTodo(initialTodoState);
      handleGetVisibilityFormState(false);
    } else {
      setError({ ...error, visible: true });
      setTimeout(() => {
        setError({ ...error, visible: false });
      }, 2500);
    }
  };

  const handleSetTags = () => {
    const newTag = tagRef.current.value;
    if (todo.tags) {
      setTodo({
        ...todo,
        tags: [...todo.tags, { tag: newTag, color: showColors.color }],
      });
    } else {
      setTodo({
        ...todo,
        ["tags"]: [{ tag: newTag, color: showColors.color }],
      });
    }
    tagRef.current.value = "";
  };

  const handleDeleteTag = (tagToDelete) => {
    const updateTags = todo.tags.filter((tag) => tag.tag !== tagToDelete);
    setTodo({ ...todo, tags: updateTags });
  };

  return (
    <div
      className="bg-form"
      onClick={(e) =>
        e.target.className === "bg-form" && handleGetVisibilityFormState(false)
      }
    >
      <form className="form">
        <div className="content-form">
          <div className="title-form">
            {itemToEdit === undefined ? "Create your task" : "Editing"}
          </div>
          <input
            className="form-title"
            value={todo.title}
            type="text"
            placeholder="Title..."
            onChange={(e) => setTodo({ ...todo, title: e.target.value })}
          />
          <select
            className="form-level"
            defaultValue={todo.level}
            onChange={(e) => setTodo({ ...todo, level: e.target.value })}
          >
            <option className="select-level">easy</option>
            <option className="select-level">medium</option>
            <option className="select-level">hard</option>
            <option className="select-level">killing</option>
            <option className="select-level">ðŸ’€</option>
          </select>
          <select
            className="form-status"
            defaultValue="-- select --"
            onChange={(e) => setTodo({ ...todo, status: e.target.value })}
          >
            <option
              className="select-level"
              value="-- select --"
              disabled
              selected
            >
              -- select --
            </option>
            <option className="select-level" value="todo">
              To Do
            </option>
            <option className="select-level" value="inProgress">
              In Progress
            </option>
            <option className="select-level" value="done">
              Completed
            </option>
          </select>
          <div
            className="start-date-selector"
            onClick={() =>
              setTimestampSelectorVisibility(!timestampSelectorVisibility)
            }
          >
            Start date: <span>{todo.date}</span>
          </div>
          {timestampSelectorVisibility && (
            <div className="timeStamp-input">
              <Calendar
                locale="en-EN"
                onChange={() => setTimestamp(Timestamp)}
                value={Timestamp}
                onClickDay={(e) => handleGetStartDate(e)}
              />
            </div>
          )}
          <textarea
            className="form-description"
            value={todo.description}
            placeholder="Description..."
            onChange={(e) => setTodo({ ...todo, description: e.target.value })}
          />
          <div className="content-tag-input">
            {showColors && showColors.visibility && (
              <CirclePicker
                className="color-selector"
                color={showColors.color}
                onClick={(e) =>
                  e.target.className !== "color-selector" &&
                  handleGetVisibilityFormState(false)
                }
                onChangeComplete={(e) =>
                  setshowColors({ visibility: false, color: e.hex })
                }
              />
            )}
            <div
              className="color-selected"
              style={{ backgroundColor: showColors.color }}
              onClick={() => {
                setshowColors({
                  ...showColors,
                  visibility: !showColors.visibility,
                });
              }}
            ></div>
            <input
              ref={tagRef}
              type="text"
              className="form-tag"
              placeholder="Write your tag here"
            />
            <i className="fas fa-plus-square icon" onClick={handleSetTags}></i>
          </div>
          <div className="content-tags">
            {todo.tags &&
              todo.tags.length > 0 &&
              todo.tags.map((tag) => {
                return (
                  <Tag
                    key={tag.tag}
                    tag={tag}
                    handleDeleteTag={handleDeleteTag}
                  />
                );
              })}
          </div>
          <Button
            size="large"
            label={itemToEdit === undefined ? "Create" : "Update"}
            onClick={(e) => handleSaveTodo(e)}
          />
          <div className="content-message">
            {error.visible && (
              <Message
                children={error.title}
                color="#ED4E2C"
                padding=".5vh 2vw"
                margin="5vh auto"
              />
            )}
          </div>
        </div>
        <i
          className="far fa-window-close icon close-icon"
          onClick={() => {
            handleGetVisibilityFormState(false);
            setTodo(initialTodoState);
          }}
        ></i>
      </form>
    </div>
  );
};
