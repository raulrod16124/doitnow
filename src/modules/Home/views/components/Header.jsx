import React, { useRef, useState } from 'react'

import { SettingsIcon } from '../../../global/Icons';
import UserSettings from './UserSettings';

export const Header = ({todos, userData}) => {

    const todosDone = todos.filter( todo => todo.status === "done" );

    const [ userSettingsVisibility, setuserSettingsVisibility ] = useState(false);

    // References 
    const bgModalWindow = useRef();
    const modalWindow = useRef();

    const handleDisplayModalWindow = () =>{
        setuserSettingsVisibility(true);
        bgModalWindow.current.style.display = "block"
        bgModalWindow.current.style.opacity = "1"
        setTimeout(()=>{
            modalWindow.current.style.width = "30%";
        },200);
    };

    const handleCloseModalWindow = () =>{
        modalWindow.current.style.width = "0%";
        bgModalWindow.current.style.opacity = "0"
        setTimeout(()=>{
            bgModalWindow.current.style.display = "none"
            setuserSettingsVisibility(false);
        },200);
    };

    return (
        <div className="header">
            <div className="header-top">
                <div className="data-of-app"></div>
                <div className="user-settings">
                    <p className="user-name">{userData && userData.body ? userData.body.username : ""}</p>
                    <div className="user-avatar">🐱‍👤</div>
                    <SettingsIcon className="icon" size="2.5" onClick={handleDisplayModalWindow} />
                </div>
                <div className="user-settings-block" ref={bgModalWindow} onClick={(e)=>{
                    if(e.target.className === "user-settings-block"){handleCloseModalWindow();}
                    }} >
                    <div className="user-settings-content" ref={modalWindow} >
                        {userSettingsVisibility && 
                            <UserSettings handleCloseModalWindow={handleCloseModalWindow} />
                        }
                    </div>
                </div>
            </div>
            <div className="header-bottom">
              <div className="progress-bar">
                  <div className="level-bar">
                      <div className="green-fill" style={{width: todos.length > 0 ? todosDone.length / todos.length * 100 + "%" : 0 + "%"}} ></div>
                  </div>
              </div>
            </div>
        </div>
    )
}
