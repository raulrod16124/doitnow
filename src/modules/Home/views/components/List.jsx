import React, { useState } from "react";
import { Droppable } from "react-beautiful-dnd";

import { Button } from "../../../../stories/Button";
import Item from "./Item";

function List({ droppableID, list, titleList, className, handleGetEditItem }) {
  // // console.log(list);

  // Extend List View Controller
  const [seeMore, setSeeMore] = useState(3);
  return (
    <Droppable droppableId={droppableID}>
      {(droppableProvided) => (
        <ul
          className={className}
          {...droppableProvided.droppableProps}
          ref={droppableProvided.innerRef}
        >
          <div className="header-list">
            <h2 className="list-title">{titleList}</h2>
            <span className="count">{Object.values(list).length}</span>
          </div>
          {Object.values(list)
            .slice(0, seeMore)
            .map((item, index) => {
              return (
                <Item
                  key={item.id}
                  item={item}
                  index={index}
                  handleGetEditItem={handleGetEditItem}
                />
              );
            })}
          {droppableProvided.placeholder}
          {Object.values(list).length > 3 && (
            <>
              {seeMore === 3 ? (
                <p className="see-more" onClick={() => setSeeMore(1000)}>
                  See more ({Object.values(list).length - 3})
                </p>
              ) : (
                <p className="see-more" onClick={() => setSeeMore(3)}>
                  See less
                </p>
              )}
            </>
          )}
        </ul>
      )}
    </Droppable>
  );
}

export default List;
