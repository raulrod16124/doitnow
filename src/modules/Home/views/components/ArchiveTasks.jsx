import React, { useEffect, useState } from "react";

import { Search } from "../../../../stories/Search";
import { ArchiveItem } from "./ArchiveItem";

export const ArchiveTasks = ({ allTodos, handleArchiveVisibility }) => {
  const [archiveTasks, setArchiveTasks] = useState([]);
  const [filterArchiveTasks, setFilterArchiveTasks] = useState([]);

  useEffect(async () => {
    setArchiveTasks(allTodos.filter((task) => task.status === "archive"));
    setFilterArchiveTasks(allTodos.filter((task) => task.status === "archive"));
    // console.log(archiveTasks);
  }, []);

  const handleSearchArchiveTask = (e) => {
    if (e.target.value === "") {
      setFilterArchiveTasks(archiveTasks);
    }
    const searchingArchiveTask = archiveTasks.filter((task) => {
      if (
        task.title.includes(e.target.value) ||
        task.description.includes(e.target.value)
      ) {
        return task;
      }
    });
    setFilterArchiveTasks(searchingArchiveTask);
  };

  return (
    <div className="bg-archive">
      <div className="archive">
        <i
          className="far fa-window-close icon close-icon"
          onClick={() => handleArchiveVisibility(false)}
        ></i>
        <div className="top-content">
          <h2 className="archive-title">
            <i className="fas fa-archive icon archive-icon-title"></i>Archived
            tasks
          </h2>
          <div className="content-search">
            <Search
              width="60"
              margin="4vmin"
              onChange={(e) => handleSearchArchiveTask(e)}
            />
          </div>
        </div>
        <div className="content-archive-tasks">
          {filterArchiveTasks.length > 0 ? (
            filterArchiveTasks.map((item, index) => {
              return (
                <ArchiveItem key={item.id + index} item={item} index={index} />
              );
            })
          ) : (
            <h3 className="empty-message">There are no archived tasks yet</h3>
          )}
        </div>
        <p className="archive-count">{archiveTasks.length} tasks</p>
      </div>
    </div>
  );
};
