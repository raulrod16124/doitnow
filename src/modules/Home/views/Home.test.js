import { mount } from 'enzyme';
import { DragDropContext } from 'react-beautiful-dnd';
import { Provider } from 'react-redux';
import { combineReducers, createStore } from 'redux';

import { Footer } from '../../global/Footer';
import { Header } from '../../global/Header';
import { TodosReducer } from "../state/reducer";
import { FormTodo } from './components/FormTodo';
import List from './components/List';
import { Home } from './Home';

describe( "Visual Tests of Home Component" , ()=>{

    let mockStore;
    let wrapper;

    beforeAll(()=>{
        const reducers = combineReducers({
            TodosReducer:
            TodosReducer,
        });
        mockStore = createStore(reducers);
        wrapper = mount( 
            <Provider store={mockStore}>
                <Home /> 
            </Provider>
        );
    });

    test('should render Home component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should find one home classname', () => {
        expect(wrapper.find('.home')).toHaveLength(1);
    })
    
    test('should find a DragDropContext component', () => {
        expect(wrapper.find(DragDropContext)).toHaveLength(1);
    })
    
    test('should find one main classname', () => {
        expect(wrapper.find('.main')).toHaveLength(1);
    })
    
    test('should find a FormTodo component', () => {
        expect(wrapper.find(FormTodo)).toHaveLength(1);
    })
    
    test('should find a Header component', () => {
        expect(wrapper.find(Header)).toHaveLength(1);
    })
    
    test('should find one content-body classname', () => {
        expect(wrapper.find('.content-body')).toHaveLength(1);
    })

    test('should find three List component', () => {
        expect(wrapper.find(List)).toHaveLength(3);
    })
    
    test('should find three Footer component', () => {
        expect(wrapper.find(Footer)).toHaveLength(1);
    })
});

describe( "Functinonality Tests of Home Component" , ()=>{
   
    let mockStore;
    let wrapper;

    beforeAll(()=>{
        const reducers = combineReducers({
            TodosReducer:
            TodosReducer,
        });
        mockStore = createStore(reducers);
        wrapper = mount( 
            <Provider store={mockStore}>
                <Home /> 
            </Provider>
        );
    });

    test('should add a new todo', () => {
        // wrapper.find(".form-title").simulate("change", { target: { value: "New todo" } });
    })
    

});