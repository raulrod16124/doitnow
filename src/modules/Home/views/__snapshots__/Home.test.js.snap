// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visual Tests of Home Component should render Home component correctly 1`] = `
<div
  className="home"
>
  <div
    className="main"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <div
        className="title-app animate__animated animate__bounceIn color1"
      >
        DO IT NOW!!
      </div>
      <input
        className="form-title"
        onChange={[Function]}
        placeholder="Title..."
        type="text"
      />
      <select
        className="form-level"
        defaultValue="easy"
        onChange={[Function]}
      >
        <option
          className="select-level"
        >
          easy
        </option>
        <option
          className="select-level"
        >
          mediun
        </option>
        <option
          className="select-level"
        >
          hard
        </option>
        <option
          className="select-level"
        >
          killing
        </option>
        <option
          className="select-level"
        >
          ðŸ’€
        </option>
      </select>
      <textarea
        className="form-description"
        onChange={[Function]}
        placeholder="Description..."
      />
      <button
        className="button button--large button--primary"
        type="submit"
      >
        Add new task
      </button>
      <div
        className="content-message"
      />
    </form>
    <div
      className="header"
    />
    <div
      className="content-body"
    >
      <div
        className="content-lists"
      >
        <ul
          className="list"
          data-rbd-droppable-context-id="0"
          data-rbd-droppable-id="todo"
        >
          <h2
            className="list-title"
          >
            TO DO
          </h2>
          <li
            aria-describedby="rbd-hidden-text-0-hidden-text-0"
            className="item"
            data-rbd-drag-handle-context-id="0"
            data-rbd-drag-handle-draggable-id="1111"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="1111"
            draggable={false}
            onDragStart={[Function]}
            onTransitionEnd={null}
            role="button"
            style={
              Object {
                "transform": null,
                "transition": null,
              }
            }
            tabIndex={0}
          >
            <div
              className="content-todo"
            >
              <div
                className="content-data-todo-left"
              >
                <h3
                  className="todo-title"
                >
                  Todo
                </h3>
                <p
                  className="todo-level"
                >
                  MODE: 
                  easy
                </p>
                <div
                  className="content-icons"
                >
                  <svg
                    className="icon"
                    fill="none"
                    height="3vmin"
                    onClick={[Function]}
                    viewBox="0 0 20 20"
                    width="3vmin"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M13.8542 3.59561C13.8541 3.59568 13.8542 3.59555 13.8542 3.59561L4.80915 12.6503L3.81363 16.189L7.35682 15.1957L16.4018 6.14C16.4746 6.06722 16.5161 5.96795 16.5161 5.86503C16.5161 5.76221 16.4753 5.6636 16.4026 5.59083C16.4025 5.59076 16.4026 5.59091 16.4026 5.59083L14.4038 3.59568C14.3309 3.52292 14.232 3.48197 14.1289 3.48197C14.026 3.48197 13.927 3.52297 13.8542 3.59561ZM12.8051 2.54754C13.1562 2.19695 13.6324 2 14.1289 2C14.6254 2 15.1016 2.19693 15.4527 2.54747C15.4527 2.5475 15.4527 2.54745 15.4527 2.54747L17.4515 4.54263C17.8026 4.89333 18 5.36914 18 5.86503C18 6.36091 17.8028 6.8365 17.4518 7.18719L8.26993 16.3799C8.17984 16.4701 8.06798 16.5356 7.94516 16.57L2.94244 17.9724C2.68418 18.0448 2.4069 17.9723 2.21725 17.7829C2.0276 17.5934 1.95512 17.3165 2.02768 17.0586L3.43296 12.0633C3.46728 11.9413 3.53237 11.8301 3.62199 11.7404L12.8051 2.54754Z"
                      fill="#676879"
                      fillRule="evenodd"
                    />
                  </svg>
                  <svg
                    className="icon"
                    fill="none"
                    height="3vmin"
                    onClick={[Function]}
                    viewBox="0 0 20 20"
                    width="3vmin"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M8.30035 1.86462C7.77994 1.86462 7.29477 2.08976 6.94732 2.46719C6.60179 2.84253 6.41724 3.33927 6.41724 3.84552V4.32642H4.901H2.63477C2.22055 4.32642 1.88477 4.6622 1.88477 5.07642C1.88477 5.49063 2.22055 5.82642 2.63477 5.82642H4.151V16.1545C4.151 16.6608 4.33556 17.1575 4.68109 17.5328C5.02853 17.9103 5.51371 18.1354 6.03411 18.1354H13.9659C14.4863 18.1354 14.9715 17.9103 15.3189 17.5328C15.6645 17.1575 15.849 16.6608 15.849 16.1545V5.82642H17.3652C17.7794 5.82642 18.1152 5.49063 18.1152 5.07642C18.1152 4.6622 17.7794 4.32642 17.3652 4.32642H15.099H13.5828V3.84552C13.5828 3.33927 13.3982 2.84253 13.0527 2.46719C12.7053 2.08976 12.2201 1.86462 11.6997 1.86462H8.30035ZM7.16447 5.82642C7.16539 5.82642 7.16631 5.82642 7.16724 5.82642H12.8328C12.8337 5.82642 12.8346 5.82642 12.8356 5.82642H14.349V16.1545C14.349 16.3012 14.2948 16.4306 14.2153 16.5169C14.1378 16.6012 14.0465 16.6354 13.9659 16.6354H6.03411C5.95348 16.6354 5.86223 16.6012 5.78468 16.5169C5.7052 16.4306 5.651 16.3012 5.651 16.1545V5.82642H7.16447ZM12.0828 4.32642V3.84552C12.0828 3.69887 12.0286 3.56943 11.9491 3.4831C11.8716 3.39886 11.7803 3.36462 11.6997 3.36462H8.30035C8.21972 3.36462 8.12847 3.39886 8.05091 3.4831C7.97144 3.56943 7.91724 3.69887 7.91724 3.84552V4.32642L12.0828 4.32642Z"
                      fill="#676879"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="content-data-todo-rigth"
              >
                <p
                  className="todo-description"
                >
                  nothing
                </p>
              </div>
            </div>
            <div
              className="bg-circle"
            />
          </li>
        </ul>
        <ul
          className="list"
          data-rbd-droppable-context-id="0"
          data-rbd-droppable-id="inProgress"
        >
          <h2
            className="list-title"
          >
            IN PROGRESS
          </h2>
          <li
            aria-describedby="rbd-hidden-text-0-hidden-text-0"
            className="item in-progress"
            data-rbd-drag-handle-context-id="0"
            data-rbd-drag-handle-draggable-id="5688"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="5688"
            draggable={false}
            onDragStart={[Function]}
            onTransitionEnd={null}
            role="button"
            style={
              Object {
                "transform": null,
                "transition": null,
              }
            }
            tabIndex={0}
          >
            <div
              className="content-todo"
            >
              <div
                className="content-data-todo-left"
              >
                <h3
                  className="todo-title"
                >
                  In Progress
                </h3>
                <p
                  className="todo-level"
                >
                  MODE: 
                  easy
                </p>
                <div
                  className="content-icons"
                >
                  <svg
                    className="icon"
                    fill="none"
                    height="3vmin"
                    onClick={[Function]}
                    viewBox="0 0 20 20"
                    width="3vmin"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M13.8542 3.59561C13.8541 3.59568 13.8542 3.59555 13.8542 3.59561L4.80915 12.6503L3.81363 16.189L7.35682 15.1957L16.4018 6.14C16.4746 6.06722 16.5161 5.96795 16.5161 5.86503C16.5161 5.76221 16.4753 5.6636 16.4026 5.59083C16.4025 5.59076 16.4026 5.59091 16.4026 5.59083L14.4038 3.59568C14.3309 3.52292 14.232 3.48197 14.1289 3.48197C14.026 3.48197 13.927 3.52297 13.8542 3.59561ZM12.8051 2.54754C13.1562 2.19695 13.6324 2 14.1289 2C14.6254 2 15.1016 2.19693 15.4527 2.54747C15.4527 2.5475 15.4527 2.54745 15.4527 2.54747L17.4515 4.54263C17.8026 4.89333 18 5.36914 18 5.86503C18 6.36091 17.8028 6.8365 17.4518 7.18719L8.26993 16.3799C8.17984 16.4701 8.06798 16.5356 7.94516 16.57L2.94244 17.9724C2.68418 18.0448 2.4069 17.9723 2.21725 17.7829C2.0276 17.5934 1.95512 17.3165 2.02768 17.0586L3.43296 12.0633C3.46728 11.9413 3.53237 11.8301 3.62199 11.7404L12.8051 2.54754Z"
                      fill="#676879"
                      fillRule="evenodd"
                    />
                  </svg>
                  <svg
                    className="icon"
                    fill="none"
                    height="3vmin"
                    onClick={[Function]}
                    viewBox="0 0 20 20"
                    width="3vmin"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M8.30035 1.86462C7.77994 1.86462 7.29477 2.08976 6.94732 2.46719C6.60179 2.84253 6.41724 3.33927 6.41724 3.84552V4.32642H4.901H2.63477C2.22055 4.32642 1.88477 4.6622 1.88477 5.07642C1.88477 5.49063 2.22055 5.82642 2.63477 5.82642H4.151V16.1545C4.151 16.6608 4.33556 17.1575 4.68109 17.5328C5.02853 17.9103 5.51371 18.1354 6.03411 18.1354H13.9659C14.4863 18.1354 14.9715 17.9103 15.3189 17.5328C15.6645 17.1575 15.849 16.6608 15.849 16.1545V5.82642H17.3652C17.7794 5.82642 18.1152 5.49063 18.1152 5.07642C18.1152 4.6622 17.7794 4.32642 17.3652 4.32642H15.099H13.5828V3.84552C13.5828 3.33927 13.3982 2.84253 13.0527 2.46719C12.7053 2.08976 12.2201 1.86462 11.6997 1.86462H8.30035ZM7.16447 5.82642C7.16539 5.82642 7.16631 5.82642 7.16724 5.82642H12.8328C12.8337 5.82642 12.8346 5.82642 12.8356 5.82642H14.349V16.1545C14.349 16.3012 14.2948 16.4306 14.2153 16.5169C14.1378 16.6012 14.0465 16.6354 13.9659 16.6354H6.03411C5.95348 16.6354 5.86223 16.6012 5.78468 16.5169C5.7052 16.4306 5.651 16.3012 5.651 16.1545V5.82642H7.16447ZM12.0828 4.32642V3.84552C12.0828 3.69887 12.0286 3.56943 11.9491 3.4831C11.8716 3.39886 11.7803 3.36462 11.6997 3.36462H8.30035C8.21972 3.36462 8.12847 3.39886 8.05091 3.4831C7.97144 3.56943 7.91724 3.69887 7.91724 3.84552V4.32642L12.0828 4.32642Z"
                      fill="#676879"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="content-data-todo-rigth"
              >
                <p
                  className="todo-description"
                >
                  nothing
                </p>
              </div>
            </div>
            <div
              className="bg-circle in-progress"
            />
          </li>
        </ul>
        <ul
          className="list"
          data-rbd-droppable-context-id="0"
          data-rbd-droppable-id="done"
        >
          <h2
            className="list-title"
          >
            DONE
          </h2>
          <li
            aria-describedby="rbd-hidden-text-0-hidden-text-0"
            className="item done"
            data-rbd-drag-handle-context-id="0"
            data-rbd-drag-handle-draggable-id="5424"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="5424"
            draggable={false}
            onDragStart={[Function]}
            onTransitionEnd={null}
            role="button"
            style={
              Object {
                "transform": null,
                "transition": null,
              }
            }
            tabIndex={0}
          >
            <div
              className="content-todo"
            >
              <div
                className="content-data-todo-left"
              >
                <h3
                  className="todo-title"
                >
                  Done
                </h3>
                <p
                  className="todo-level"
                >
                  MODE: 
                  easy
                </p>
                <div
                  className="content-icons"
                >
                  <svg
                    className="icon"
                    fill="none"
                    height="3vmin"
                    onClick={[Function]}
                    viewBox="0 0 20 20"
                    width="3vmin"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M13.8542 3.59561C13.8541 3.59568 13.8542 3.59555 13.8542 3.59561L4.80915 12.6503L3.81363 16.189L7.35682 15.1957L16.4018 6.14C16.4746 6.06722 16.5161 5.96795 16.5161 5.86503C16.5161 5.76221 16.4753 5.6636 16.4026 5.59083C16.4025 5.59076 16.4026 5.59091 16.4026 5.59083L14.4038 3.59568C14.3309 3.52292 14.232 3.48197 14.1289 3.48197C14.026 3.48197 13.927 3.52297 13.8542 3.59561ZM12.8051 2.54754C13.1562 2.19695 13.6324 2 14.1289 2C14.6254 2 15.1016 2.19693 15.4527 2.54747C15.4527 2.5475 15.4527 2.54745 15.4527 2.54747L17.4515 4.54263C17.8026 4.89333 18 5.36914 18 5.86503C18 6.36091 17.8028 6.8365 17.4518 7.18719L8.26993 16.3799C8.17984 16.4701 8.06798 16.5356 7.94516 16.57L2.94244 17.9724C2.68418 18.0448 2.4069 17.9723 2.21725 17.7829C2.0276 17.5934 1.95512 17.3165 2.02768 17.0586L3.43296 12.0633C3.46728 11.9413 3.53237 11.8301 3.62199 11.7404L12.8051 2.54754Z"
                      fill="#676879"
                      fillRule="evenodd"
                    />
                  </svg>
                  <svg
                    className="icon"
                    fill="none"
                    height="3vmin"
                    onClick={[Function]}
                    viewBox="0 0 20 20"
                    width="3vmin"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M8.30035 1.86462C7.77994 1.86462 7.29477 2.08976 6.94732 2.46719C6.60179 2.84253 6.41724 3.33927 6.41724 3.84552V4.32642H4.901H2.63477C2.22055 4.32642 1.88477 4.6622 1.88477 5.07642C1.88477 5.49063 2.22055 5.82642 2.63477 5.82642H4.151V16.1545C4.151 16.6608 4.33556 17.1575 4.68109 17.5328C5.02853 17.9103 5.51371 18.1354 6.03411 18.1354H13.9659C14.4863 18.1354 14.9715 17.9103 15.3189 17.5328C15.6645 17.1575 15.849 16.6608 15.849 16.1545V5.82642H17.3652C17.7794 5.82642 18.1152 5.49063 18.1152 5.07642C18.1152 4.6622 17.7794 4.32642 17.3652 4.32642H15.099H13.5828V3.84552C13.5828 3.33927 13.3982 2.84253 13.0527 2.46719C12.7053 2.08976 12.2201 1.86462 11.6997 1.86462H8.30035ZM7.16447 5.82642C7.16539 5.82642 7.16631 5.82642 7.16724 5.82642H12.8328C12.8337 5.82642 12.8346 5.82642 12.8356 5.82642H14.349V16.1545C14.349 16.3012 14.2948 16.4306 14.2153 16.5169C14.1378 16.6012 14.0465 16.6354 13.9659 16.6354H6.03411C5.95348 16.6354 5.86223 16.6012 5.78468 16.5169C5.7052 16.4306 5.651 16.3012 5.651 16.1545V5.82642H7.16447ZM12.0828 4.32642V3.84552C12.0828 3.69887 12.0286 3.56943 11.9491 3.4831C11.8716 3.39886 11.7803 3.36462 11.6997 3.36462H8.30035C8.21972 3.36462 8.12847 3.39886 8.05091 3.4831C7.97144 3.56943 7.91724 3.69887 7.91724 3.84552V4.32642L12.0828 4.32642Z"
                      fill="#676879"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="content-data-todo-rigth"
              >
                <p
                  className="todo-description"
                >
                  nothing
                </p>
              </div>
            </div>
            <div
              className="bg-circle done"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <p
      className="footer-text"
    >
      Created by RaÃºl RodrÃ­guez
    </p>
    <p
      className="footer-copy-right"
    >
      All rights reserved Â©2021
    </p>
  </footer>
</div>
`;
